### PR TITLE
Add io.github.umair_khurshid.caesar_cipher

### PR DESCRIPTION
**Pull request by Umair-khurshid**
### Please confirm your submission meets all the criteria

- [X] Description: Caesar Cipher is a simple CLI utility for encoding and decoding text using the Caesar Cipher technique. 
- [X] The domain used for the application ID is [controlled by the application developer(s)][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor to the project. If not, I contacted upstream developers about this submission. **Link:** N/A (I am the developer).

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by hfiguiere** [comment]
> Please describe the application briefly. 

Still missing

File: .flatpak-builder/ccache/disabled/ccache.conf
Comment (hfiguiere) [inline .flatpak-builder/ccache/disabled/ccache.conf:1]: this doesn't belong here.

**Review by hfiguiere — CHANGES_REQUESTED** [review]
none of the comment from the previous PR were addressed.

File: caesarcipher.png
Comment (hfiguiere) [inline caesarcipher.png:1]: this should be upstream

File: io.github.umair_khurshid.caesar_cipher.metainfo.xml
Comment (hfiguiere) [inline io.github.umair_khurshid.caesar_cipher.metainfo.xml:1]: this should be upstream

File: io.github.umair_khurshid.caesar_cipher.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  <categories>
+    <category>System</category>
+    <category>Security</category>
+  </categories>
+
+  <provides>
+    <binary>caesar</binary>
+  </provides>
+</component>
```
Comment (hfiguiere) [inline io.github.umair_khurshid.caesar_cipher.metainfo.xml:36]: missing mandatory screenshot, release

File: io.github.umair_khurshid.caesar_cipher.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  <metadata_license>MIT</metadata_license>
+  <project_license>MIT</project_license>
+
+  <description>
+    <p>
+      A simple Caesar cipher encryption tool written in C. The Caesar cipher is one of the simplest and oldest methods of encryption.
+    </p>
+  </description>
+
+  <icon type="stock">caesarcipher</icon>
```
Comment (hfiguiere) [inline io.github.umair_khurshid.caesar_cipher.metainfo.xml:17]: don't use stock icons

```suggestion
```

File: io.github.umair_khurshid.caesar_cipher.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+modules:
+  - name: caesar-cipher
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/bin
+      - gcc -o caesar caesar.c
+      - install caesar /app/bin/
+    sources:
+      - type: git
+        url: https://github.com/Umair-khurshid/caesar-cipher.git
```
Comment (hfiguiere) [inline io.github.umair_khurshid.caesar_cipher.yml:25]: you need a specific revision. not putting a branch is like putting one.

File: io.github.umair_khurshid.caesar_cipher.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+  <content_rating type="oars-1.1" />
+
+  <developer id="com.umair">
```
Comment (hfiguiere) [inline io.github.umair_khurshid.caesar_cipher.metainfo.xml:21]: it doesn't look like `umair.com` is you.

File: io.github.umair_khurshid.caesar_cipher.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+id: io.github.UmairKhurshid.CaesarCipher
+runtime: org.freedesktop.Platform
+runtime-version: "24.08"
+sdk: org.freedesktop.Sdk
+command: caesar
+    
+tags:
+  - CLI
+  - Utilities
```
Comment (hfiguiere) [inline io.github.umair_khurshid.caesar_cipher.yml:9]:
```suggestion
```

but I still don't see which part of the doc suggested these.

File: io.github.umair_khurshid.caesar_cipher.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+finish-args:
+  - --share=network
```
Comment (hfiguiere) [inline io.github.umair_khurshid.caesar_cipher.yml:11]: it's hard to tell but does this really require network?

File: io.github.umair_khurshid.caesar_cipher.yml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  - --filesystem=home
```
Comment (hfiguiere) [inline io.github.umair_khurshid.caesar_cipher.yml:12]: does this require filesystem access?

File: io.github.umair_khurshid.caesar_cipher.yml [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  - --socket=fallback-x11
+  - --socket=wayland
+
+modules:
+  - name: caesar-cipher
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/bin
+      - gcc -o caesar caesar.c
```
Comment (hfiguiere) [inline io.github.umair_khurshid.caesar_cipher.yml:21]: no build system?

File: io.github.umair_khurshid.caesar_cipher.yml [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+      - install caesar /app/bin/
```
Comment (hfiguiere) [inline io.github.umair_khurshid.caesar_cipher.yml:22]:
```suggestion
      - gcc -o caesar caesar.c
      - install -Dm755 caesar -t /app/bin/
```

**Comment by bbhtt** [comment]
Please remove the commits from the master branch of this repo and rebase this to new-pr branch properly.

**Pull request closed without merging**